## 1. Load the Data

The first step is to read the CSV file into pandas so the data can be explored and analyzed. This allows the airline arrival information to be structured into a DataFrame for easier comparison.

In [1]:
import pandas as pd

df = pd.read_csv("Project01.csv")

df

,Airline,Status,Los Angeles,Phoenix,San Diego,San Francisco,Seattle
0,ALASKA,on time,497,221,212,503,1841
1,NaN,delayed,62,12,20,102,305
2,AM WEST,on time,694,4840,383,320,201
3,NaN,delayed,117,415,65,129,61


## Clean up Repeated Airline Labels

Some airline names are blank in rows where the label should carry down.  
I’m using forward fill so every row keeps the right airline name.

**Key point:** this makes grouping by airline possible later. (easier)

In [2]:
df["Airline"] = df["Airline"].ffill()

df

,Airline,Status,Los Angeles,Phoenix,San Diego,San Francisco,Seattle
0,ALASKA,on time,497,221,212,503,1841
1,ALASKA,delayed,62,12,20,102,305
2,AM WEST,on time,694,4840,383,320,201
3,AM WEST,delayed,117,415,65,129,61


## Focus on Delayed Flights Only

Now I’m filtering to just rows where `Status == "delayed"`.

**Key point:** I've isolated the records we need for delay comparisons.

In [3]:
delays = df[df["Status"] == "delayed"]

delays

,Airline,Status,Los Angeles,Phoenix,San Diego,San Francisco,Seattle
1,ALASKA,delayed,62,12,20,102,305
3,AM WEST,delayed,117,415,65,129,61


## 4. Calculating Total Delays

The next step is to calculate the `Total Delays` of flights across all five destinations for each airline. Summing these values provides a direct comparison of overall arrival delays.

**Key point:** this creates a single number we can aggregate by airline.

In [4]:
delays["Total Delays"] = delays.iloc[:, 2:].sum(axis=1)

delays[["Airline", "Total Delays"]]


,Airline,Total Delays
1,ALASKA,501
3,AM WEST,787


## 5. Comparing Delay Rates

The raw delay totals help, but rates are fairer since airlines may have different flight volumes.  
Here I calculate total flights, total delays, then delay rate percentage.

**Key point:** delay rate gives the more balanced comparison.


In [5]:
df["Total Flights"] = df.iloc[:, 2:].sum(axis=1)

totalFlights = df.groupby("Airline")["Total Flights"].sum()
totalDelays = delays.groupby("Airline")["Total Delays"].sum()

delayRate = (totalDelays / totalFlights) * 100

delayRate


Airline
ALASKA     13.271523
AM WEST    10.892734
dtype: float64

## Final Thoughts

After cleaning the provided data, I focused on delayed arrivals to compare performance between the two airlines. AM WEST had a higher total number of delays across the five destinations with `787 total delays`. However, looking at delay rates instead of just raw totals gave a clearer picture of reliability. When the number of delayed flights is considered in relation to total flights, the comparison feels more grounded. Overall, the analysis suggests that proportional performance tells a more meaningful story than simple counts alone.